src/geo/leaflet/leaflet-layer-view-factory.js

```diff
@@ -1,24 +1,60 @@
+var _ = require('underscore');
 var log = require('cdb.log');
 var LeafletTiledLayerView = require('./leaflet-tiled-layer-view');
 var LeafletWMSLayerView = require('./leaflet-wms-layer-view');
 var LeafletPlainLayerView = require('./leaflet-plain-layer-view');
 var LeafletCartoDBLayerGroupView = require('./leaflet-cartodb-layer-group-view');
 var LeafletTorqueLayerView = require('./leaflet-torque-layer-view');
 var LeafletCartoDBWebglLayerGroupView = require('./leaflet-cartodb-webgl-layer-group-view');
+var TangramCartoCSS = require('tangram-cartocss');
+var RenderModes = require('../../vis/render-modes');
 
-var LayerGroupViewConstructor = function (layerGroupModel, mapModel, options) {
-  if (options.vector) {
-    return new LeafletCartoDBWebglLayerGroupView(layerGroupModel, mapModel);
+var MAX_NUMBER_OF_FEATURES_FOR_WEBGL = 10e4;
+
+var LayerGroupViewConstructor = function (layerGroupModel, nativeMap, mapModel, settingsModel) {
+  if (canMapBeRenderedClientSide(mapModel, settingsModel)) {
+    return new LeafletCartoDBWebglLayerGroupView(layerGroupModel, nativeMap);
+  }
+
+  return new LeafletCartoDBLayerGroupView(layerGroupModel, nativeMap);
+};
+
+var canMapBeRenderedClientSide = function (mapModel, settingsModel) {
```

Comment:
@donflopez could you please review latest changes to this logic? I feel like it should be extracted into it's own module / class so we can test it in isolation (it's currently untested)... what do you think? thanks!